classes/Customer.php

```diff
@@ -197,7 +197,11 @@ class CustomerCore extends ObjectModel
 
     public function __construct($id = null)
     {
-        $this->id_default_group = (int)Configuration::get('PS_CUSTOMER_GROUP');
```

Comment:
@LittleBigDev "Customer's constructor is not a good place to define it's group." but as you can see right here, the customer's group is defined in the construct ;)